### PR TITLE
Remove collection dependency from gather node

### PR DESCRIPTION
**Pull request by gnusi**
Please note that for legal reasons we require you to sign the 
[Contributor Agreement](https://www.arangodb.com/documents/cla.pdf)
before we can accept your pull requests.

### CODE REVIEW

**Comment by gnusi** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr-linux/

**Comment by gnusi** [comment]
Corresponding enterprise PR:
https://github.com/arangodb/enterprise/pull/100

File: arangod/Aql/ClusterNodes.h [hunk 1]
```diff
@@ -372,44 +397,17 @@ class GatherNode final : public ExecutionNode {
...
   SortElementVector _elements;
 
-  /// @brief the underlying database
-  TRI_vocbase_t* _vocbase;
-
-  /// @brief the underlying collection
-  Collection const* _collection;
-
-  /// @brief (optional) auxiliary collections (satellites)
-  std::unordered_set<Collection const*> _auxiliaryCollections;
```
Comment (graetzer) [inline arangod/Aql/ClusterNodes.h]: are these actually unused?!

**Comment by jsteemann** [comment]
passed: http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr-linux/561/

**Pull request merged**